### PR TITLE
Create release Workflow

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
the release workflow creates a release on MapStore, builds the proper artifacts and upload them. 
This speeds up the release process and do not need jenkins. 

TODO: 
- [ ] The only thing missing is the docker publishing. 

From Github UI one run it :

https://github.com/offtherailz/MapStore2/actions/workflows/release.yml

image

To create a release like this draft, on the branch inserted: 

https://github.com/offtherailz/MapStore2/releases/tag/untagged-7b8e9d7e350e8e5c6cc8

image

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [x] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#<issue>

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by tdipisa** [comment]
@randomorder we have this PR still in draft that would allow effectively to replace the release Jenkins job with a github action. It is only missing of the docker publishing. A review/feedback from you would be great?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: "cache maven dependencies"
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: mapstore-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            mapstore-${{ runner.os }}-maven-
```
Comment (offtherailz) [inline .github/workflows/release.yml:59]:
```suggestion
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        required: true
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "greet"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    steps:
+      ############
```
Comment (offtherailz) [inline .github/workflows/release.yml:27]:
```suggestion
      - name: "checking out"
        uses: actions/checkout@v2
        with: 
          ref: ${{ github.event.inputs.branch }}
      - name: "setting up npm"
        uses: actions/setup-node@v2
        with:
            node-version: '12.x'
      - name: "setting up Java"
        uses: actions/setup-java@v1
        with:
          java-version: '8.x'
      ############
      # CACHING
      ##########
      - name: "cache node modules"
        uses: actions/cache@v1
        env:
          cache-name: cache-node-modules
        with:
          path: ~/.npm # npm cache files are stored in `~/.npm` on Linux/macOS
          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package.json') }}
          restore-keys: |
            ${{ runner.os }}-build-${{ env.cache-name }}-
            ${{ runner.os }}-build-
            ${{ runner.os }}-
      - name: "cache maven dependencies"
        uses: actions/cache@v1
        with:
          path: ~/.m2/repository
          key: mapstore-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: |
            mapstore-${{ runner.os }}-maven-
      ############
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          body: |
+            ## [${{ github.event.inputs.version }}](https://github.com/geosolutions-it/MapStore2/tree/v${{ github.event.inputs.version }}) (${{ steps.date.outputs.date }})
+
+              - **[Full Changelog](https://github.com/geosolutions-it/MapStore2/compare/v${{ github.event.inputs.previousVersion }}...v${{ github.event.inputs.version }})**
+
+              - **[Implemented enhancements](https://github.com/geosolutions-it/MapStore2/issues?q=is%3Aissue+milestone%3A%22${{ github.event.inputs.version }}%22+is%3Aclosed+label%3Aenhancement)**
+
+              - **[Fixed bugs](https://github.com/geosolutions-it/MapStore2/issues?q=is%3Aissue+milestone%3A%22${{ github.event.inputs.version }}%22+is%3Aclosed+label%3Abug)**
+
+              - **[Closed issues](https://github.com/geosolutions-it/MapStore2/issues?q=is%3Aissue+milestone%3A%22${{ github.event.inputs.version }}%22+is%3Aclosed)**
```
Comment (offtherailz) [inline .github/workflows/release.yml:113]:
```suggestion
            ## Useful links related to [v${{ github.event.inputs.version }}](https://github.com/geosolutions-it/MapStore2/tree/v${{ github.event.inputs.version }})
            
            - [Full Changelog](<https://github.com/geosolutions-it/MapStore2/compare/v${{ github.event.inputs.previousVersion }}...v${{ github.event.inputs.version }})
            
            - [Implemented enhancements](https://github.com/geosolutions-it/MapStore2/issues?q=is%3Aissue+milestone%3A%22${{ github.event.inputs.version }}%22+is%3Aclosed+label%3Aenhancement)
            
            - [Fixed bugs](https://github.com/geosolutions-it/MapStore2/issues?q=is%3Aissue+milestone%3A%22${{ github.event.inputs.version }}%22+is%3Aclosed+label%3Abug)
            
            - [Closed issues](https://github.com/geosolutions-it/MapStore2/issues?q=is%3Aissue+milestone%3A%22${{ github.event.inputs.version }}%22+is%3Aclosed)
            
            - [MapStore Extension release v${{ github.event.inputs.version }}](https://github.com/geosolutions-it/MapStoreExtension/releases/tag/v${{ github.event.inputs.version }})
            
            - [Docker image v${{ github.event.inputs.version }}](https://hub.docker.com/layers/geosolutionsit/mapstore2/${{ github.event.inputs.version }}/images/sha256-a2ad2e451c13fa2ae5a011f8163b9ab9025257e2caa4e6a63546973c3b0b9fa9?context=explore)
            
            - [MapStore documentation v${{ github.event.inputs.version }}](https://docs.mapstore.geosolutionsgroup.com/en/v${{ github.event.inputs.version }}/)
```

**Comment by offtherailz** [comment]
Closing in favor of #9555

**Pull request closed without merging**